lib/rspec/mocks/method_double.rb

```diff
@@ -80,14 +90,16 @@ def proxy_method_invoked(_obj, *args, &block)
 
       # @private
       def restore_original_method
-        return show_frozen_warning if object_singleton_class.frozen?
         return unless @method_is_proxied
 
         remove_method_from_definition_target
         @method_stasher.restore if @method_stasher.method_is_stashed?
         restore_original_visibility
 
         @method_is_proxied = false
+      rescue RuntimeError => e
+        return show_frozen_warning if FROZEN_ERROR_MSG === e.message
```

Comment:
Rather than checking the message which will change from time to time why not use the frozen check here? Or are you saying Rails has broken that as well?

```suggestion
        return show_frozen_warning if object_singleton_class.frozen?
```

Reply:
In ruby 2.5 there's a specific exception class for frozen error, so the long term goal would be to use that rather than the (as noted) changeable message.

But yes, I thought it would be good if you could unstub "frozen" rails objects as well as stub them.

Reply:
I thought of a way to improve this, added another commit for it. I'll also look at putting together a different PR to clean this up on the 4.x branch (I think moving to ruby 2.5+ was mentioned)